driver-mapping/src/main/java/com/datastax/driver/mapping/MethodMapper.java

```diff
@@ -92,6 +88,30 @@ public void prepare(MappingManager manager, PreparedStatement ps) {
         }
     }
 
+    // Checks the method parameters against the query's bind variables
+    private void validateParameters() {
+        if (method.isVarArgs())
+            throw new IllegalArgumentException(String.format("Invalid varargs method %s in @Accessor interface", method.getName()));
+
+        ColumnDefinitions variables = statement.getVariables();
+        Set<String> names = Sets.newHashSet();
+        for (ColumnDefinitions.Definition variable : variables) {
+            names.add(variable.getName());
+        }
+
+        if (method.getParameterTypes().length < names.size())
```

Comment:
There is an edge case with anonymous markers. Cassandra will name the marker after the column, but a column can legitimately appear twice in a query with different values.

For example this statement has two distinct names, but you'll want to bind 3 values:
```java
PreparedStatement pst = session.prepare("update foo set v = ? where k = ? if v = ?");
Set<Object> names = Sets.newHashSet();
for (ColumnDefinitions.Definition variable : pst.getVariables()) {
    names.add(variable.getName());
}
System.out.println(names.size());
```